Update Profiler UI for Symfony 6.2+.

nickygerritsen opened the pull request:
This is based on the closed PR #1558, but only does two changes:

* It updates the icon in 6.2+.
* It adds the metric groups.

I will do a separate PR to add the tabs.

Screenshot on 6.1:
<img width="1310" alt="image" src="https://user-images.githubusercontent.com/550145/210865401-fdae5616-9c9b-43ee-b1c7-1333e4aac4bf.png">

And on 6.2:
<img width="1246" alt="image" src="https://user-images.githubusercontent.com/550145/210865433-b98a0b67-1aaa-4867-bb2d-e54c11e0fbcb.png">

nickygerritsen commented:
Not sure why the test fails btw. If I add `$twigLoaderFilesystem->addPath(__DIR__ . '/../Resources/views', 'Doctrine');` to the `setUp` method it works, but we already load other resources in the twig...?

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -426,7 +486,9 @@
             header.setAttribute('data-sort-direction', (-1*direction).toString());
-            header.children[0].innerHTML = direction > 0 ? '<span class="text-muted">&#9650;</span>' : '<span class="text-muted">&#9660;</span>';
+            header.children[0].innerHTML = direction > 0
+                ? `<span class="sort-icon">{{ include('@Doctrine/Collector/arrow-narrow-up.svg') }}</span>`
```
Would be nice to know why are we replacing UTF-8 character with SVG, which is larger and harder to change.

nickygerritsen replied:
I think because the UTF-8 characters don't look nice. But I also didn't really like the arrows, so I leave this to someone with actual UX/UI skills.

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -98,289 +107,340 @@
-                    <p>No loaded entities.</p>
-                </div>
-            {% else %}
-                <table>
```
Not sure why are we changing position of this section, this makes the PR quite hard to review

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -98,289 +107,340 @@
 
-    {% set group_queries = request.query.getBoolean('group') %}
-    {% if group_queries %}
-        <h2>Grouped Statements</h2>
```
why are these headings removed?

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -98,289 +107,340 @@
+                <span class="label">Different statements</span>
+            </div>
 
-        <div class="metric">
```
Why was this moved under? Seems quite random change to me.

nickygerritsen replied:
Agreed, reverted.

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -98,289 +107,340 @@
             </div>
+        {% endif %}
+
+        {% if collector.invalidEntityCount %}
```
This also seem quite arbitrary decision to me to hide the section when count is 0

nickygerritsen replied:
Agreed, got rid of this.

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -8,7 +8,11 @@
             {% set status = collector.invalidEntityCount > 0 ? 'red' : collector.querycount > 50 ? 'yellow' %}
 
-            {{ include('@Doctrine/Collector/icon.svg') }}
+            {% if profiler_markup_version < 3 %}
```
these can be simplified like so
```twig
<span class="icon">{{ include('@Doctrine/Collector/' ~ (profiler_markup_version < 3 ? 'icon' : 'database') ~ '.svg') }}</span>
```

nickygerritsen commented:
I changed the PR quite a bit and only updated the icon and added the metric groups. Once this is merged we can work from there to add the tabs for the different things.

I also didn't drop any metrics or changed any order anymore.

ostrolucky commented on Resources/views/Collector/db.html.twig:
```diff
@@ -103,38 +103,42 @@
-            <span class="value">{{ collector.invalidEntityCount }}</span>
-            <span class="label">Invalid entities</span>
-        </div>
+        <div class="metric-group">
```
What are these for? Is it so that it stretches to full width?

nickygerritsen replied:
It “groups” the metrics. Look at the screenshots: in 6.1 it is 4 blocks, in 6.2 it is 4 parts of a bigger block. Without it, there would be whitespace between the blocks
